src/modules/simulation/gz_bridge/GZMixingInterfaceServo.cpp

```diff
@@ -0,0 +1,95 @@
+ *
+ *   Copyright (c) 2023 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *	notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *	notice, this list of conditions and the following disclaimer in
+ *	the documentation and/or other materials provided with the
+ *	distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *	used to endorse or promote products derived from this software
+ *	without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "GZMixingInterfaceServo.hpp"
+
+bool GZMixingInterfaceServo::init(const std::string &model_name)
+{
+#if 0 // TODO
+	// output eg /X500/command/motor_speed
+	std::string actuator_topic = "/" + model_name + "/command/motor_speed";
+	_actuators_pub = _node.Advertise<ignition::msgs::Actuators>(actuator_topic);
+
+	if (!_actuators_pub.Valid()) {
+		PX4_ERR("failed to advertise %s", actuator_topic.c_str());
+		return false;
+	}
+
+#endif
+
+	return true;
+}
+
+bool GZMixingInterfaceServo::updateOutputs(bool stop_motors, uint16_t outputs[MAX_ACTUATORS], unsigned num_outputs,
+		unsigned num_control_groups_updated)
+{
+
+	// cmd.command_value = (float)outputs[i] / 500.f - 1.f; // [-1, 1]
+
+#if 0 // TODO
+	unsigned active_output_count = 0;
+
+	for (unsigned i = 0; i < num_outputs; i++) {
+		if (_mixing_output.isFunctionSet(i)) {
+			active_output_count++;
+
+		} else {
+			break;
+		}
+	}
+
+	if (active_output_count > 0) {
+		ignition::msgs::Actuators rotor_velocity_message;
+		rotor_velocity_message.mutable_velocity()->Resize(active_output_count, 0);
+
+		for (unsigned i = 0; i < active_output_count; i++) {
+			rotor_velocity_message.set_velocity(i, outputs[i]);
+		}
+
+		if (_actuators_pub.Valid()) {
+			return _actuators_pub.Publish(rotor_velocity_message);
+		}
+	}
+
+#endif
```

Comment:
I guess this is still left as to do -> but where would the joint angle scaling happen?

Reply:
In here, but at the moment the outputs PX4 side need to be positive to fit into a uint16, I'm thinking later we should change that so that they can be configured with the native units.